packages/create-redwood-app/template/api/src/lib/auth.ts

```diff
@@ -11,7 +11,12 @@ export const isAuthenticated = () => {
   return true
 }
 
-export const hasRole = ({ roles }) => {
+/**
+ * When checking role membership, roles can be a single value, a list, or none.
+ */
+type AllowedRoles = string | string[] | undefined
+
+export const hasRole = (roles: AllowedRoles) => {
```

Comment:
I think it's more accurate to how this function is actually used to change the types to this.

```suggestion
type AllowedRoles = string | string[]

export const hasRole = (roles?: AllowedRoles) => {
```

But I know that's not how it looks in the other auth.ts templates for when authentication is actually implemented... So feel free to ignore this comment. Or fix the other templates too.

Reply:
Hmmm, i'm undecided. Having undefined in there explicitly has something to it as there is an actual `=== undefined`-check in the method – but on the other hand, less code is always better code ;)

@dthyresson came up with that type in 10a1b066b47888e607f7aad3a01c456f306f9ef4, so maybe let them chime in here.

I can live with leaving it like it is, b/c, again, that would change the scope of this PR.

Reply:
I'll hand it over to you as i'll be off until Monday, so anyone is welcome to push onto here and go ahead with this.

Reply:
> Hmmm, i'm undecided. Having undefined in there explicitly has something to it

It's not a stylistic preference. The two options are actually different. See here https://www.typescriptlang.org/play?#code/C4TwDgpgBAggNnA9gdwgEwEqLhAzlAXil2ACcBLAOwHMoAfYsq6gbQF16oBXStCAMyroAUMNCRYCFOiw5cAOUQBVXgMKMKNTiU2s2ogMaJKJKAAsAhrlnQiAClLY8ALklJUmJ7gCUhAHxQAN7CUFBGJk4AdEjUdgDkjnJxADRQiXjewgC+hsamltZOAIrqDl4A-K7w7jJeiip8-L4EAcGh4bhRMfHpuClpXpk5wgU2dpmjxeOikzh2PI1CaBNWNkXzqoKU6N5AA

Reply:
> @dthyresson came up with that type in https://github.com/redwoodjs/redwood/commit/10a1b066b47888e607f7aad3a01c456f306f9ef4, so maybe let them chime in here – there may have been a good reason to define it exactly that way.

I am ok with it being optional as we factor Auth.